jetty-io/src/main/java/org/eclipse/jetty/io/EndPoint.java

```diff
@@ -203,7 +203,15 @@ default void close()
      * filled or -1 if EOF is read or the input is shutdown.
      * @throws IOException if the endpoint is closed.
      */
-    int fill(ByteBuffer buffer) throws IOException;
+    default int fill(ByteBuffer buffer) throws IOException
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    default SocketAddress receive(ByteBuffer buffer) throws IOException
+    {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
I'm not convinced it is correct to add this method to our abstract EndPoint.  OK for the short term, but perhaps we should consider refactoring so that `EndPoint` has neither `fill` nor `receive`.  Instead these signatures should be moved to `DatagramEndPoint` and `StreamEndPoint`?   Ditto for `flush`,  `write` and `send`.

But let's leave it this way for now and just be aware that it may need to change.